src/test/javascript/spec/component/drag-and-drop-question/drag-and-drop-question.component.spec.ts

```diff
@@ -0,0 +1,228 @@
+
+    it('should hide sample solutions', () => {
+        comp.showingSampleSolution = true;
+        comp.hideSampleSolution();
+        expect(comp.showingSampleSolution).to.equal(false);
+    });
+
+    it('should return unassigned drag items', () => {
+        const [_1, mapping1, dragItem1] = getDropLocationMappingAndItem();
+        const [_2, _m2, dragItem2] = getDropLocationMappingAndItem();
+        comp.mappings = [mapping1];
+        comp.question.dragItems = [dragItem1, dragItem2];
+        expect(comp.getUnassignedDragItems()).to.deep.equal([dragItem2]);
+    });
+
+    it('should return invalid dragItem for location', () => {
+        const [dropLocation1, mapping1] = getDropLocationMappingAndItem();
+        const [dropLocation2, mapping2, dragItem2] = getDropLocationMappingAndItem();
+        comp.mappings = [mapping1, mapping2];
+        expect(comp.invalidDragItemForDropLocation(dropLocation1)).to.equal(false);
+        dragItem2.invalid = true;
+        expect(comp.invalidDragItemForDropLocation(dropLocation2)).to.equal(true);
+    });
+
+    it('should return no drag item if there is no mapping', () => {
+        const [dropLocation] = getDropLocationMappingAndItem();
+        expect(comp.dragItemForDropLocation(dropLocation)).to.null;
+    });
+
+    it('should remove existing mappings when there is no drop location', () => {
+        const [_, mapping, dragItem] = getDropLocationMappingAndItem();
+        const [_1, mapping1] = getDropLocationMappingAndItem();
+        const [_2, mapping2] = getDropLocationMappingAndItem();
+        checkDragDrop(undefined, dragItem, [mapping, mapping1], [mapping1], 1);
+
+        // should not call update if mappings did not change
+        checkDragDrop(undefined, dragItem, [mapping1, mapping2], [mapping1, mapping2], 0);
+    });
+
+    it('should not do anything if given droplocation and dragEvent dragData is mapped', () => {
+        const [dropLocation, mapping, dragItem] = getDropLocationMappingAndItem();
+        checkDragDrop(dropLocation, dragItem, [mapping], [mapping], 0);
+    });
+
+    it('should map dragItem to new drop location', () => {
+        const [dropLocation, mapping, dragItem] = getDropLocationMappingAndItem();
+        const [dropLocation1, mapping1, dragItem1] = getDropLocationMappingAndItem();
+        const [_, mapping2] = getDropLocationMappingAndItem();
+        const newMappings = [mapping2, new DragAndDropMapping(dragItem1, dropLocation), new DragAndDropMapping(dragItem, dropLocation1)];
+        checkDragDrop(dropLocation, dragItem1, [mapping, mapping1, mapping2], newMappings, 1);
+    });
+
+    const checkDragDrop = (
+        dropLocation: DropLocation | undefined,
+        dragItem: DragItem,
+        mappings: DragAndDropMapping[],
+        expectedMappings: DragAndDropMapping[],
+        callCount: number,
+    ) => {
+        comp.mappings = mappings;
+        const dragEvent = { dragData: dragItem };
+        let fnOnMappingUpdate = sinon.fake();
```

Comment:
GitHub Actions complains that this should be `const` instead of `let`.

Reply:
Fixed it